python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -96,7 +96,7 @@ export default function EditorContainer({
   const onSave = useCallback(async () => {
     setIsSaving(true);
     try {
-      await callbacks.save(clientConfigToAIConfig(aiconfigState));
+      await callbacks.save(clientConfigToAIConfig(stateRef.current));
```

Comment:
Why wouldn't we want to re-render if client aiconfig changes?

Reply:
We will always re-render if aiconfigState changes. This stateRef is a stable ref to use in the callbacks since otherwise using aiconfigState in the callback means the callback logic is actually using the state from when the callback was constructed & memoized (which could technically be stale)

Reply:
Ah ok this makes sense!

Reply:
I don't necessarily know if it had to be update here, but it was a small enough change and related to using proper source of truth (like id instead of index)